pkg/asset/cluster/tfvars.go

```diff
@@ -39,7 +38,7 @@ func (t *TerraformVariables) Dependencies() []asset.Asset {
 func (t *TerraformVariables) Generate(parents map[asset.Asset]*asset.State) (*asset.State, error) {
 	installCfg, err := installconfig.GetInstallConfig(t.installConfig, parents)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get install config state in the parent asset states")
+		return nil, errors.Wrap(err, "failed to get InstallConfig from parents")
```

Comment:
This may be overdoing it.  For example, simulating an error with:

```diff
diff --git a/pkg/asset/installconfig/installconfig.go b/pkg/asset/installconfig/installconfig.go
index 33a7833..578afda 100644
--- a/pkg/asset/installconfig/installconfig.go
+++ b/pkg/asset/installconfig/installconfig.go
@@ -154,7 +154,7 @@ func GetInstallConfig(installConfig asset.Asset, parents map[asset.Asset]*asset.
 	var cfg types.InstallConfig
 
 	st, ok := parents[installConfig]
-	if !ok {
+	if !ok || true {
 		return nil, errors.Errorf("%T does not exist in parents", installConfig)
 	}
 
```

this leads to:

```console
$ openshift-install cluster
FATAL Failed to generate Terraform Variables: failed to fetch dependency for Terraform Variables: failed to fetch dependency for Bootstrap Ignition Config: failed to generate asset Certificate (): failed to get InstallConfig from parents: *installconfig.installConfig does not exist in parents 
```

I think those last two (`failed to get InstallConfig from parents: *installconfig.installConfig does not exist in parents`) could be collapsed down to a single `*installconfig.installConfig does not exist in parents` or `failed to get InstallConfig from parents`, but I don't think we need both.

Reply:
I think `failed to get InstallConfig from parents: *installconfig.installConfig does not exist in parents` seems like its giving you correct context that it couldn't get installconfig because *it doesn't exist*.

Reply:
> I think `failed to get InstallConfig from parents: *installconfig.installConfig does not exist in parents` seems like its giving you correct context that it couldn't get installconfig because *it doesn't exist*.

And you think:

    FATAL Failed to generate Terraform Variables: failed to fetch dependency for Terraform Variables: failed to fetch dependency for Bootstrap Ignition Config: failed to generate asset Certificate (): *installconfig.installConfig does not exist in parents

would fail to convey that?

Reply:
https://github.com/openshift/installer/blob/6295cec6e8db9171a0b73ae404b0b7f5608700e9/pkg/asset/installconfig/installconfig.go#L153-L166

has two error situations. Since we are wrapping the errors:
```
_, err := GetInstallConfig(parents)
if err != nil {
    return errros.Wrap(err, "failed to get InstallConfig from parents")
}
```
Creates:
`failed to get InstallConfig from parents: *installconfig.installConfig does not exist in parents` and
`failed to get InstallConfig from parents: failed to unmarshal the installconfig`

Personally I think that makes sense.